src/utils/resource_stat.py

```diff
@@ -174,52 +218,61 @@ def __imul__(self, other):
                 self_v = self.res.get(k, 0)
                 other_v = other.res.get(k, 0)
                 self.res[k] = self_v * other_v
+
+            self.normalize()
             return self
 
     def __truediv__(self, other):
         if isinstance(other, numbers.Number):
             result = copy.deepcopy(self)
             for k, v in self.res.items():
-                result.res[k] = v / other
+                # Division by zero gives zero
+                if other == 0:
+                    result.res[k] = 0
+                else:
+                    result.res[k] = v / other
+
+            result.normalize()
             return result
         else:
             if self.__class__ != other.__class__:
                 raise ValueError("Incompatible class %s and %s" %
                                  (self.__class__, other.__class__))
 
-            if self.unit != other.unit:
-                raise ValueError("Incompatible resource type %s and %s" %
-                                 (self.unit, other.unit))
-
             # Pairwise division
             self_keys = set(self.res.keys())
             other_keys = set(other.res.keys())
             all_keys = self_keys.union(other_keys)
 
-            result = self.__class__(res=None, unit=self.unit)
+            result = self.__class__()
             for k in all_keys:
                 self_v = self.res.get(k, 0)
                 other_v = other.res.get(k, 0)
+                # Division by zero gives zero
                 if other_v == 0:
-                    raise ValueError("Division by zero. %s / %s" %
-                                     (self, other))
-                result.res[k] = self_v / other_v
+                    result.res[k] = 0
+                else:
+                    result.res[k] = self_v / other_v
+
+            result.normalize()
             return result
 
     def __idiv__(self, other):
         if isinstance(other, numbers.Number):
             for k, v in self.res.items():
-                self.res[k] = v / other
+                # Division by zero gives zero
+                if other == 0:
```

Comment:
do we need to log here and other div functions? because we are essentially swallowing the exception